### PR TITLE
Fix regression for loc and __setitem__ when one-dimensional tuple was given for MultiIndex

### LINKED ISSUE #37711: BUG: errors with `.loc[(slice(...), ), ]`  when modifying a subset of rows in a pandas dataframe/series in 1.1.4
I often use `.loc[(slice(...),),]`  for selection and modifying when dealing with both series and dataframe: see 
[advanced-indexing-with-hierarchical-index](https://pandas.pydata.org/pandas-docs/stable/user_guide/advanced.html#advanced-indexing-with-hierarchical-index)
> In [43]: df.loc['bar']
Out[43]: 
               A         B         C
second                              
one     0.895717  0.410835 -1.413681
two     0.805244  0.813850  1.607920

> This is a shortcut for the slightly more verbose notation df.loc[('bar',),] (equivalent to df.loc['bar',] in this example).

Every thinkg is OK in `0.25.4`,  but in `1.1.4` (also `1.0.3` and `1.1.3`), only selection is valid  and modifying will raise errors:

```python
import pandas as pd
s = pd.Series([1,2,3], index=pd.MultiIndex.from_tuples([('a','A'),  ('b','B'), ('c', 'C')]))
df = s.to_frame()
print(s.loc[('a', ), ])
print(df.loc[('a', ), ])
```
but when I try to modify a subset of rows using `.loc[(slice(...),),]` 
```python
In [2]: df.loc[('a', ), ] = 0
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-2-480fc1763dde> in <module>
----> 1 df.loc[('a', ), ] = 0

~/opt/anaconda3/envs/edge/lib/python3.8/site-packages/pandas/core/indexing.py in __setitem__(self, key, value)                                                                    
    664         else:
    665             key = com.apply_if_callable(key, self.obj)
--> 666         indexer = self._get_setitem_indexer(key)
    667         self._has_valid_setitem_indexer(key)
    668

~/opt/anaconda3/envs/edge/lib/python3.8/site-packages/pandas/core/indexing.py in _get_setitem_indexer(self, key)                                                                  
    591         """
    592         if self.name == "loc":
--> 593             self._ensure_listlike_indexer(key)
    594
    595         if self.axis is not None:

~/opt/anaconda3/envs/edge/lib/python3.8/site-packages/pandas/core/indexing.py in _ensure_listlike_indexer(self, key, axis)                                                        
    645             # key may be a tuple if we are .loc
    646             # in that case, set key to the column part of key
--> 647             key = key[column_axis]
    648             axis = column_axis
    649

IndexError: tuple index out of range
```
and
```python
In [3]: s.loc[('a', ), ] = 0
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-3-c3a2e38a0417> in <module>
----> 1 s.loc[('a', ), ] = 0

~/opt/anaconda3/envs/edge/lib/python3.8/site-packages/pandas/core/indexing.py in __setitem__(self, key, value)
    668 
    669         iloc = self if self.name == "iloc" else self.obj.iloc
--> 670         iloc._setitem_with_indexer(indexer, value)
    671 
    672     def _validate_key(self, key, axis: int):

~/opt/anaconda3/envs/edge/lib/python3.8/site-packages/pandas/core/indexing.py in _setitem_with_indexer(self, indexer, value)
   1633         if take_split_path:
   1634             # Above we only set take_split_path to True for 2D cases
-> 1635             assert self.ndim == 2
   1636             assert info_axis == 1
   1637 

AssertionError:
```

Note that “partial” indexing `.loc[slice(...)]` is still valid in ``1.1.4`:
```python
In [4]: df.loc['a'] = 0

In [5]: s.loc['a'] = 0

In [6]: df
Out[6]: 
     0
a A  0
b B  2
c C  3

In [7]: s
Out[7]: 
a  A    0
b  B    2
c  C    3
dtype: int64
```
Of course, `.loc[('a', )]` is valid for series  and  `.loc[('a', ), : ]`  is vaild for dataframe, but I have to use different codes for series and dataframe.

Update on 20201113:
-----------------------
- For series, `s.loc[('a')] = 0`, `s.loc[('a', )] = 0` and  `s.loc[('a'), ] = 0` are vaild,  but `s.loc[('a', ), ] = 0` will raise similar AssertionError like above.
- For dataframe, `df.loc[('a')] = 0`, `df.loc[('a'), :] = 0` and  `df.loc[('a',), :] = 0` are vaild, but `df.loc[('a', )] = 0` , `df.loc[('a'), ] = 0` and `df.loc[('a', ), ] = 0` will raise similar IndexError like above.

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #37711
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 1]
```diff
@@ -288,6 +288,20 @@ def convert_nested_indexer(indexer_type, keys):
 
         tm.assert_series_equal(result, expected)
 
+    def test_multiindex_loc_one_dimensional_tuple(self):
```
Comment (jbrockmendel) [inline pandas/tests/indexing/multiindex/test_loc.py:291]: use frame_or_series

**Comment by jbrockmendel** [comment]
@phofl you're opening PRs faster than they can be reviewed (which is awesome, keep up the good work!).  Maybe you can tackle one of the more challenging and immediate issues: the darns windows builds

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/indexing/multiindex/test_loc.py]: Done

**Comment by phofl** [comment]
Thanky very much @jbrockmendel Have a bit of free time this week.

Are you referring to the pipeline windows py38_no18? Unfortunately I am not that familiar with CI and even less with CI on windows.

File: pandas/core/indexing.py [hunk 2]
```diff
@@ -650,11 +650,12 @@ def _ensure_listlike_indexer(self, key, axis=None, value=None):
...
             return
 
-        if isinstance(key, tuple):
+        if isinstance(key, tuple) and not isinstance(self.obj.index, ABCMultiIndex):
             # key may be a tuple if we are .loc
-            # in that case, set key to the column part of key
+            # if index is not a MultiIndex, set key to column part
             key = key[column_axis]
             axis = column_axis
+            pass
```
Comment (jreback) [inline pandas/core/indexing.py:658]: extraneious pass here

**Review by jreback — CHANGES_REQUESTED** [review]
tiny comment, ping on green.

File: doc/source/whatsnew/v1.2.0.rst [hunk 3]
```diff
@@ -470,6 +470,7 @@ Indexing
 - Bug in :meth:`Index.where` incorrectly casting numeric values to strings (:issue:`37591`)
 - Bug in :meth:`Series.loc` and :meth:`DataFrame.loc` raises when numeric label was given for object :class:`Index` although label was in :class:`Index` (:issue:`26491`)
 - Bug in :meth:`DataFrame.loc` returned requested key plus missing values when ``loc`` was applied to single level from :class:`MultiIndex` (:issue:`27104`)
+- Fixed regression in :meth:`DataFrame.loc` and :meth:`Series.loc` for ``__setitem__`` when one-dimensional tuple was given to select from :class:`MultiIndex` (:issue:`37711`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:473]: actually this can go in 1.1.5 (backport might be a bit of work)

Comment (phofl) [reply doc/source/whatsnew/v1.2.0.rst]: Was not sure if there would be a 1.1.5 release. Move the whatsnew entry.

**Review by jreback — APPROVED** [review]

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 4]
```diff
@@ -288,6 +288,14 @@ def convert_nested_indexer(indexer_type, keys):
 
         tm.assert_series_equal(result, expected)
 
+    def test_multiindex_loc_one_dimensional_tuple(self, frame_or_series):
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_loc.py:291]: can you parameterize over all of the selectors in the OP e.g. `obj.loc[('a',), :]` if its a DataFrame

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 4 above)
Comment (phofl) [reply pandas/tests/indexing/multiindex/test_loc.py:291]: Added tests for df, I think one of the cases mentioned there should raise and not work

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 5]
```diff
@@ -288,6 +288,23 @@ def convert_nested_indexer(indexer_type, keys):
...
+        # GH#37711
+        mi = MultiIndex.from_tuples([("a", "A"), ("b", "A")])
+        obj = frame_or_series([1, 2], index=mi)
+        obj.loc[("a",)] = 0
+        expected = frame_or_series([0, 2], index=mi)
+        tm.assert_equal(obj, expected)
+
+    @pytest.mark.parametrize("indexer", [("a",), ("a")])
+    def test_multiindex_one_dimensiona_tuple_columns(self, indexer):
```
Comment (jbrockmendel) [inline pandas/tests/indexing/multiindex/test_loc.py:300]: typo dimensional?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm ex @jbrockmendel comment, ping on green.

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 5 above)
Comment (phofl) [reply pandas/tests/indexing/multiindex/test_loc.py:300]: Yep, thx. Fixed

**Comment by jreback** [comment]
thanks @phofl keep em coming!

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Comment by jbrockmendel** [comment]
>> you're opening PRs faster than they can be reviewed (which is awesome, keep up the good work!). Maybe you can tackle one of the more challenging and immediate issues: the darns windows builds

> Are you referring to the pipeline windows py38_no18? Unfortunately I am not that familiar with CI and even less with CI on windows.

CI is just another learning curve we all have to get up at some point. i still regularly ask for help with it.

As for windows, im not actually convinced anyone uses it ever.  It might be some kind of cruel prank played by a trickster god.  see #37746 where i only-half-kidding suggest we drop windows altogether.

More generally, I'm suggesting you look for areas that will have a higher ratio of (time to create PR) / (time to review PR).\*

\* Though im really excited about the indexing fixes you've been making.

**Pull request merged**